cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -277,6 +351,82 @@ class ndview : public ndarray_base<axis_count, order> {
     bool data_is_mutable_;
 };
 
+#ifdef ONEDAL_DATA_PARALLEL
+template <typename T1, ndorder ord1, typename T2, ndorder ord2>
+sycl::event copy_by_value(sycl::queue& q,
+                          ndview<T1, 2, ord1>& dst,
+                          const ndview<T2, 2, ord2>& src,
+                          const event_vector& deps = {}) {
+    ndshape<2> dst_shape = dst.get_shape();
+    ONEDAL_ASSERT(dst_shape == src.get_shape());
+    ONEDAL_ASSERT(dst.has_mutable_data() && src.has_data());
+    const auto cp_range = make_range_2d(dst_shape[0], dst_shape[1]);
+    if constexpr (ord1 == ndorder::c) {
+        T1* const dst_ptr = dst.get_mutable_data();
+        const T2* const src_ptr = src.get_data();
+        const auto dst_stride = dst.get_leading_stride();
+        const auto src_stride = src.get_leading_stride();
+        return q.submit([&](sycl::handler& h) {
+            h.depends_on(deps);
+            h.parallel_for(cp_range, [=](sycl::id<2> idx) {
+                T1& dst_ref = *(dst_ptr + idx[0] * dst_stride + idx[1]);
+                if constexpr (ord2 == ndorder::c) {
+                    dst_ref = static_cast<T1>(*(src_ptr + idx[0] * src_stride + idx[1]));
+                }
+                else {
+                    dst_ref = static_cast<T2>(*(src_ptr + idx[1] * src_stride + idx[0]));
+                }
+            });
+        });
+    }
+    else {
+        auto new_dst = dst.t();
+        const auto new_src = src.t();
+        return copy_by_value(q, new_dst, new_src, deps);
+    }
+    ONEDAL_ASSERT(false);
+    return sycl::event();
+}
+
+template <typename T>
+sycl::event fill_with_value(sycl::queue& q,
+                            ndview<T, 1>& dst,
+                            const T& value = T{},
+                            const event_vector& deps = {}) {
+    ONEDAL_ASSERT(dst.has_mutable_data());
+    return q.submit([&](sycl::handler& cgh) {
+        cgh.depends_on(deps);
+        cgh.fill(dst.get_mutable_data(), value, dst.get_count());
+    });
+}
+
+template <typename T, ndorder ord1>
+sycl::event fill_with_value(sycl::queue& q,
```

Comment:
fill_with_value -> fill to be aligned with ndarray

Reply:
fixed